src/NuGet.Core/NuGet.PackageManagement/Utility/PackagesConfigLockFileUtility.cs

```diff
@@ -57,7 +57,7 @@ public static class PackagesConfigLockFileUtility
                     var lockFileRelativePath = projectUri.MakeRelativeUri(lockFileUri).OriginalString;
                     if (Path.DirectorySeparatorChar != '/')
                     {
-                        lockFileRelativePath.Replace('/', Path.DirectorySeparatorChar);
+                        lockFileRelativePath = lockFileRelativePath.Replace('/', Path.DirectorySeparatorChar);
                     }
                     msbuildProject.ProjectSystem.AddExistingFile(lockFileRelativePath);
```

Comment:
If this code was really working already, perhaps msbuild can handle the separator difference internally and we can remove the entire if condition?